mantidproject/mantid
# Refactor QWorkspaceDockView

This PR covers the refactoring of the `QWorkspaceDockView` to a `QWorkspaceWidget`. This is now a self contained Widget which is not stored within a DockWidget. It no longer has reliance on it's parent class.

Additionally, `MantidUI` and has been updated to hold a QDockWidget container for the QWorkspaceWidget.

Finally the class name has been changed (as have references to this class elsewhere in the code base).

**To test:**

* Code review 

* Open `MantidPlot`
* Observe that the WorkspaceWindow is the same as it was before.
  * Check that it can still be docked and undocked
  * Check the buttons still work as expected
  * Check size limits

Fixes #21461 

**Release Notes** 
*Internal change not required in release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
